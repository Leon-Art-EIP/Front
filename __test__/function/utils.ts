import type { ReactTestRendererJSON } from "react-test-renderer";
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";
import { expect } from "vitest";

function toJson(component: renderer.ReactTestRenderer): ReactTestRendererJSON {
  const result = component.toJSON();
  expect(result).toBeDefined();

  return result as ReactTestRendererJSON;
}

export function testSnapshot(element: JSX.Element): void {
  const component = renderer.create(element);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
}
