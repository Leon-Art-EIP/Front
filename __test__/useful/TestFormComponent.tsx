import { FormProvider } from "react-hook-form";
import type { ReactNode } from "react";
import useTestForm from "./useTestForm";

interface ITestInputProps {
  name: string;
  value?: string[] | string;
  children: ReactNode;
}

export default function TestFormComponent(props: ITestInputProps): JSX.Element {
  const methods = useTestForm({ [props.name]: props.value });

  return <FormProvider {...methods}>{props.children}</FormProvider>;
}
