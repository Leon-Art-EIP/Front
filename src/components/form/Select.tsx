import { useController } from "react-hook-form";
import { IOption } from "../../interfaces";
import { cn } from "../../tools/cn";

interface ISelectProps {
  title?: string;
  name: string;
  options: IOption<string>[];
  placeholder?: string;
  className?: string;
}

export default function Select(props: ISelectProps): JSX.Element {
  const {
    field: { onChange, value },
    fieldState: { error },
  } = useController({ name: props.name });

  return (
    <div className={cn("flex flex-col gap-2", props.className)}>
      <div className="text-lg">{props.title}</div>
      <select
        name={props.name}
        id={props.name}
        onChange={onChange}
        value={value}
        className="border-none focus:outline-none p-2 rounded hover:cursor-pointer"
      >
        <option value="" disabled selected>
          {props.placeholder}
        </option>
        {props.options.map((option) => (
          <option key={option.value} value={option.value}>
            {option.label}
          </option>
        ))}
      </select>
      {error && <div className="text-red-600">{error.message}</div>}
    </div>
  );
}
