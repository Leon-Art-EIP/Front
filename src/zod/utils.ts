export const getValueOrUndefined = (arg: unknown) =>
  arg === "" || arg === undefined || arg === null ? undefined : arg;
