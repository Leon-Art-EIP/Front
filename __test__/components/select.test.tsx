import { test } from "vitest";
import { testSnapshot } from "../function/utils";
import TestFormComponent from "../useful/TestFormComponent";
import Select from "../../src/components/form/Select";

test("Select snapshot", () => {
  testSnapshot(
    <TestFormComponent name="options">
      <Select
        name="options"
        options={[
          {
            label: "Option 1",
            value: "option1",
            selected: true,
          },
          {
            label: "Option 2",
            value: "option2",
            selected: false,
          },
        ]}
      />
    </TestFormComponent>
  );
});
