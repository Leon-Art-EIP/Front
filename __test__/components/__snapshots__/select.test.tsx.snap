// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Select snapshot 1`] = `
<div
  className="flex flex-col gap-2"
>
  <select
    className="border-none focus:outline-none p-2 rounded hover:cursor-pointer"
    id="options"
    name="options"
    onChange={[Function]}
  >
    <option
      disabled={true}
      value=""
    />
    <option
      value="option1"
    >
      Option 1
    </option>
    <option
      value="option2"
    >
      Option 2
    </option>
  </select>
</div>
`;
